framework/decode/file_processor.cpp

```diff
@@ -876,25 +845,111 @@ ParsedBlock BlockParser::ParseBlock(BlockBuffer& block_buffer)
     GFXRECON_ASSERT(block_buffer.ReadPos() == sizeof(format::BlockHeader));
     const format::BlockHeader& block_header = block_buffer.Header();
     format::BlockType          base_type    = format::RemoveCompressedBlockBit(block_header.type);
-
+    ParsedBlock parsed_block = ParsedBlock(ParsedBlock::EmptyBlockTag(), ParsedBlock::BlockState::kUnknown);
     switch (base_type)
     {
         case format::kFunctionCallBlock:
-            return ParseFunctionCall(block_buffer);
+            parsed_block = ParseFunctionCall(block_buffer);
+            break;
         case format::kMethodCallBlock:
-            return ParseMethodCall(block_buffer);
+            parsed_block = ParseMethodCall(block_buffer);
+            break;
         case format::kMetaDataBlock:
-            return ParseMetaData(block_buffer);
+            parsed_block = ParseMetaData(block_buffer);
+            break;
         case format::kFrameMarkerBlock:
-            return ParseFrameMarker(block_buffer);
+            parsed_block = ParseFrameMarker(block_buffer);
+            break;
         case format::kStateMarkerBlock:
-            return ParseStateMarker(block_buffer);
+            parsed_block = ParseStateMarker(block_buffer);
+            break;
         case format::kAnnotation:
-            return ParseAnnotation(block_buffer);
+            parsed_block = ParseAnnotation(block_buffer);
+            break;
         case format::kUnknownBlock:
+            break;
         default:
-            return ParsedBlock(ParsedBlock::EmptyBlockTag(), ParsedBlock::BlockState::kUnknown);
+            break;
     }
+
+    // Support various Decompression policies, also checks if decompression is needed
+    if (DecompressWhenParsed(parsed_block))
+    {
+        GFXRECON_ASSERT(format::IsBlockCompressed(block_buffer.Header().type));
+        DecompressionVisitor decompressor(*this);
+        parsed_block.Visit(decompressor);
+    }
+
+    return parsed_block;
+}
+
+// The parameter buffer always takes up the end of the block
+// Default value for uncompressed_size is kReadSizeFromBuffer -- which is the behavior of Function and method call
+BlockParser::ParameterReadResult
+BlockParser::ReadParameterBuffer(const char* label, BlockBuffer& block_buffer, uint64_t uncompressed_size)
+{
+    ParameterReadResult result; // result.success initializes to true;
```

Comment:
Thank you for this comment.  I would have complained about the else below without it.